util/src/main/java/org/killbill/billing/util/entity/dao/EntitySqlDaoWrapperInvocationHandler.java

```diff
@@ -210,16 +211,20 @@ public Object execute() throws Throwable {
     }
 
     private Object invokeWithAuditAndHistory(final Audited auditedAnnotation, final Method method, final Object[] args) throws Throwable {
-        final InternalCallContext context = retrieveContextFromArguments(args);
+        final InternalCallContext contextMaybeWithoutAccountRecordId = retrieveContextFromArguments(args);
         final List<String> entityIds = retrieveEntityIdsFromArguments(method, args);
-
+        // We cannot always infer the TableName from the signature
+        TableName tableName = retrieveTableNameFromArgumentsIfPossible(args);
         final ChangeType changeType = auditedAnnotation.value();
+        final boolean isBatchQuery = method.getAnnotation(SqlBatch.class) != null;
 
         // Get the current state before deletion for the history tables
-        final Map<String, M> deletedEntities = new HashMap<String, M>();
+        final Map<Long, M> deletedEntities = new HashMap<Long, M>();
         if (changeType == ChangeType.DELETE) {
             for (final String entityId : entityIds) {
-                deletedEntities.put(entityId, sqlDao.getById(entityId, context));
+                // TODO Switch to getByIds
```

Comment:
This seems reasonable, but do we even have use case of delete operations in batch?

Reply:
We don't indeed - just making the code forward compatible 😆